src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -253,231 +261,110 @@ private void CheckAnalysisWasRequested()
+            mockSolution.Reset();
+            mockSolution
+                .Setup(x => x.FindProjectItem(mockedJavascriptDocumentFooJs.Name))
+                .Returns((ProjectItem)null);
 
-            testSubject.Factory.CurrentSnapshot.TryGetValue(0, StandardTableKeyNames.ProjectGuid, out var projectGuid).Should().BeTrue();
-            projectGuid.Should().Be(projectGuid);
+            Action act = () => CreateTextBufferIssueTracker();
+            act.Should().NotThrow();
         }
 
         [TestMethod]
-        public void WhenNewIssuesAreFound_AndFilterRemovesAllIssues_ListenersAreUpdated()
+        public void WhenFileIsRenamed_FileHasNoAssociatedProject_NoExceptionIsThrown()
         {
-            mockSonarErrorDataSource.Invocations.Clear();
-
-            // Arrange
-            var filteredIssue = CreateIssue("single issue", startLine: 1, endLine: 1);
+            mockSolution.Reset();
+            mockSolution
+                .Setup(x => x.FindProjectItem(mockedJavascriptDocumentFooJs.Name))
+                .Returns((ProjectItem)null);
 
-            var issuesToReturnFromFilter = Enumerable.Empty<IAnalysisIssueVisualization>();
-            SetupIssuesFilter(out var capturedFilterInput, issuesToReturnFromFilter);
+            Action act = () => mockedJavascriptDocumentFooJs.Raise(x => x.FileActionOccurred += null, new TextDocumentFileActionEventArgs(
+                mockedJavascriptDocumentFooJs.Name, DateTime.Now,
+                FileActionTypes.DocumentRenamed));
 
-            // Act
-            testSubject.HandleNewIssues(new [] { filteredIssue });
+            act.Should().NotThrow();
 
-            // Assert
-            // Check the expected issues were passed to the filter
-            capturedFilterInput.Count.Should().Be(1);
-            capturedFilterInput[0].Should().Be(filteredIssue);
+            logger.AssertNoOutputMessages();
+        }
 
-            CheckErrorListRefreshWasRequestedOnce(testSubject.Factory);
+        #endregion
 
-            // Check there are no issues
-            testSubject.Factory.CurrentSnapshot.Issues.Count().Should().Be(0);
-        }
+        #region RequestAnalysis
 
         [TestMethod]
-        public void WhenNewFileLevelIssuesAreFound_IssueNotFiltered_ListenersAreUpdated()
+        public void Create_AnalysisIsRequestedOnCreation()
         {
-            mockSonarErrorDataSource.Invocations.Clear();
-
-            // Arrange
-            var issues = new[] { CreateFileLevelIssue("File Level Issue") };
+            var issueConsumer = new Mock<IIssueConsumer>();
+            var issueConsumerFactory = CreateIssueConsumerFactory(issueConsumer.Object);
 
-            SetupIssuesFilter(out var capturedFilterInput, issues);
+            testSubject = CreateTextBufferIssueTracker(issueConsumerFactory: issueConsumerFactory.Object);
 
-            // Act
-            testSubject.HandleNewIssues(issues);
+            // The factory should have been called...
```

Comment:
The comment is slightly unhelpful/redundant, as it's exactly what the verify is doing. Perhaps it would help to be more explicit "verify that a new issue consumer was created"